application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -62,4 +70,19 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         }
         message.queue();
     }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        if (event.getName().equals(COMMAND_NAME)
+                && event.getFocusedOption().getName().equals(ID_OPTION)) {
+            List<Command.Choice> choices = tagSystem.getAllIds()
+                .stream()
+                .filter(id -> id.startsWith(event.getFocusedOption().getValue()))
```

Comment:
I'd instead refactor the StringDistance method to get *good* autocompletion instead of a simple startsWith

Reply:
i don't get what u mean, can u elaborate

Reply:
There's a StringDistance class or something, take a look at that class.

Reply:
can't find any

Reply:
Look again, StringDistances

IntelliJ has a feature which allows you to look for classes, you cannot miss it when using that

Reply:
![2022-10-01_115204-751x578_scrot](https://user-images.githubusercontent.com/73871477/193395795-359bb72b-b65b-4d69-9848-5e43eef514b5.png)

Reply:
In IntelliJ I said, maybe it's a class in our project (yes it is :p

Reply:
ok found it

Reply:
it won't work i believe it only returns a single item not a collection

Reply:
startsWith should be just fine

Reply:
Pretty sure I tagged you in the Discord about this.

Add an autocompletion method which does the same as autocomplete, but then it returns a collection

Reply:
It is "fine" but I much rather have better
See my messages in the Discord